core/kernel/panic.c

```diff
@@ -4,11 +4,57 @@
  * Copyright (c) 2014, STMicroelectronics International N.V.
  */
 
+#include <initcall.h>
+#include <kernel/interrupt.h>
+#include <kernel/misc.h>
 #include <kernel/panic.h>
 #include <kernel/thread.h>
 #include <kernel/unwind.h>
 #include <trace.h>
 
+static_assert(!IS_ENABLED(CFG_HALT_CORES_ON_PANIC) ||
```

Comment:
An inline comment is welcome to explain why SGI number should be in that range.
Something like `/* SGI number must be in the secure SGI range */`